src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/MasterEventMetricsSnapshot.java

```diff
@@ -0,0 +1,311 @@
+     *
+     * CurrentWindowEndTime: 1535065341025
+     * Actual Table
+     * |tid  |insertOrder|taskType    |priority|queueTime|metadata|           st|           et|
+     * +-----+-----------+------------+--------+---------+--------+-------------+-------------+
+     * |111  |1          |create-index|urgent  |3        |{string}|1535065340625|       {null}|
+     * |111  |2          |create-index|urgent  |12       |{string}|1535065340825|       {null}|
+     * |111  |1          |      {null}|  {null}|   {null}|  {null}|       {null}|1535065340725|
+     *
+     * Returned:
+     *
+     * |tid  |insertOrder|taskType    |priority|queueTime|metadata|           st|           et|
+     * +-----+-----------+------------+--------+---------+--------+-------------+-------------+
+     * |111  |1          |create-index|urgent  |3        |{string}|1535065340625|1535065340725|
+     * |111  |2          |create-index|urgent  |12       |{string}|1535065340825|1535065341025|
+     *
+     * @return aggregated master task
+     */
+    public SelectHavingStep<Record> groupByInsertOrderWithoutNull() {
+
+        Long endTime = windowStartTime + MetricsConfiguration.SAMPLING_INTERVAL;
+        List<SelectField<?>> fields = getGroupByInsertOrderSelectFields();
+        fields.add(DSL.least(DSL.coalesce(DSL.max(DSL.field(Fields.ET.toString(), Long.class)), endTime), endTime)
+                .as(DSL.name(Fields.ET.toString())));
+
+        ArrayList<Field<?>> groupByInsertOrder = new ArrayList<Field<?>>() { {
+            this.add(DSL.field(DSL.name(AllMetrics.Master_Metric_Dimensions.MASTER_TASK_INSERT_ORDER.toString()), String.class));
+        } };
+
+        return create.select(fields)
+                .from(DSL.table(this.tableName))
+                .groupBy(groupByInsertOrder);
+    }
+
+    /**
+     * Return one row per master task event. Group by the InsertOrder, with possible et remains as null
+     *
+     * Actual Table
+     * |tid  |insertOrder|taskType    |priority|queueTime|metadata|           st|           et|
+     * +-----+-----------+------------+--------+---------+--------+-------------+-------------+
+     * |111  |1          |create-index|urgent  |3        |{string}|1535065340625|       {null}|
+     * |111  |2          |create-index|urgent  |12       |{string}|1535065340825|       {null}|
+     * |111  |1          |      {null}|  {null}|   {null}|  {null}|       {null}|1535065340725|
+     *
+     * Returned:
+     *
+     * |tid  |insertOrder|taskType    |priority|queueTime|metadata|           st|           et|
+     * +-----+-----------+------------+--------+---------+--------+-------------+-------------+
+     * |111  |1          |create-index|urgent  |3        |{string}|1535065340625|1535065340725|
+     * |111  |2          |create-index|urgent  |12       |{string}|1535065340825|       {null}|
+     *
+     * @return aggregated latency rows for each shard request
+     */
+    public SelectHavingStep<Record> groupByInsertOrder() {
```

Comment:
If Elasticsearch is in a crashloop, we will get multiple tasks with the same insert order. What do you think the right behavior is in this case?